pims/bioformats.py

```diff
@@ -1,36 +1,102 @@
-from __future__ import (absolute_import, division, print_function)
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
 
 import numpy as np
 
 from pims.base_frames import FramesSequence
 from pims.frame import Frame
 from warnings import warn
-from os.path import isfile
+import os
 
 try:
-    import javabridge
+    import jpype
 except ImportError:
-    javabridge = None
+    jpype = None
 
-try:
-    import bioformats
-except ImportError:
-    bioformats = None
 
-try:
-    from pandas import DataFrame
-except ImportError:
-    DataFrame = None
+def available():
+    return jpype is not None
 
 
-def available():
-    try:
-        import javabridge
-        import bioformats
-    except ImportError:
-        return False
+def _gen_jar_locations():
+    """
+    Generator that yields optional locations of loci_tools.jar.
+    The precedence order is (highest priority first):
+
+    1. pims package location
+    2. /etc/loci_tools.jar
+    3. CONDA_ENV/etc/loci_tools.jar
+    4. PROGRAMDATA/pims/loci_tools.jar
+    5. LOCALAPPDATA/pims/loci_tools.jar
+    6. APPDATA/pims/loci_tools.jar
+    7. ~/.config/pims/loci_tools.jar
+    """
+    yield os.path.dirname(__file__)
+    yield '/etc'
+    if 'CONDA_ETC_' in os.environ:
```

Comment:
This isn't a canonical thing, this is needs https://github.com/NSLS-II/conda-prescriptions/tree/master/releases/conda_etc to work.